### PR TITLE
Added Metal 2 Port and Metal 2 Service Profile Tests

### PR DESCRIPTION
**Pull request by jkallem-equinix**

### CODE REVIEW

File: examples/metal-nimf-2-port-connection/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+output "metal_vlan_id" {
+  value = equinix_metal_vlan.vlan-server.id
+}
+output "metal_connection_id" {
+  value = equinix_metal_connection.metal-connection.id
+}
+output "metal_port_connection_id" {
+  value = module.metal_2_port_connection.primary_connection_id
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-port-connection/outputs.tf:9]: nit: Add extra space

File: examples/metal-nimf-2-port-connection/terraform.tfvars.example [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+notifications_emails        = ["example@equinix.com"]
+bandwidth                   = 50
+purchase_order_number       = "1-323292"
+project_id                  = "<Project_ID>"
+zside_ap_type               = "COLO"
+zside_port_name             = "ops-user100-CX-SV1-NL-Qinq-STD-1G-PRI-NK-349"
+zside_vlan_tag              = "3711"
+zside_location              = "SV"
+
+
```
Comment (srushti-patl) [inline examples/metal-nimf-2-port-connection/terraform.tfvars.example:25]: nit: Remove extra space

File: examples/metal-nimf-2-port-connection/variables.tf [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  type        = string
+}
+variable "zside_port_name" {
+  description = "Equinix Port Name"
+  type        = string
+}
+variable "zside_vlan_tag" {
+  description = "Vlan Tag information, outer vlanSTag for QINQ connections"
+  type        = string
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-port-connection/variables.tf:89]: nit: Add extra space

File: examples/metal-nimf-2-service-profile-connection/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Metal to SERVICE_PROFILE Connection Example
```
Comment (srushti-patl) [inline examples/metal-nimf-2-service-profile-connection/README.md:1]: Service profile should be written like this: Generic Service Profile. Replace these words in the entire README.md file.

File: examples/metal-nimf-2-service-profile-connection/main.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  purchase_order_number = var.purchase_order_number
+
+  aside_ap_authentication_key = equinix_metal_connection.metal-connection.authorization_code
+
+  zside_ap_type         = var.zside_ap_type
+  zside_ap_profile_type = var.zside_ap_profile_type
+  zside_location        = var.zside_location
+  zside_fabric_sp_name  = var.zside_fabric_sp_name
+}
+
```
Comment (srushti-patl) [inline examples/metal-nimf-2-service-profile-connection/main.tf:41]: nit:remove extra space

File: examples/metal-nimf-2-service-profile-connection/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,9 @@
+output "metal_vlan_id" {
+  value = equinix_metal_vlan.vlan-server.id
+}
+output "metal_connection_id" {
+  value = equinix_metal_connection.metal-connection.id
+}
+output "metal_service_profile_connection_id" {
+  value = module.metal_2_service_profile.primary_connection_id
+}
```
Comment (srushti-patl) [inline examples/metal-nimf-2-service-profile-connection/outputs.tf:9]: nit: Add extra space

File: examples/metal-nimf-2-service-profile-connection/variables.tf [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+variable "zside_fabric_sp_name" {
+  description = "Equinix Service Profile Name"
+  type        = string
+}
+variable "zside_seller_region" {
+  description = "Access point seller region"
+  type        = string
+}
+variable "zside_location" {
+  description = ""
```
Comment (srushti-patl) [inline examples/metal-nimf-2-service-profile-connection/variables.tf:91]: Add zside_location description

**Review by srushti-patl — CHANGES_REQUESTED** [review]
Requested few changes.

**Review by srushti-patl — APPROVED** [review]
LGTM. Waiting for @thogarty's approval.

File: examples/metal-nimf-2-port-connection/versions.tf [hunk 8]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.5.4"
+  required_providers {
+    equinix = {
+      source  = "equinix/equinix"
+      version = ">= 1.34.0"
```
Comment (thogarty) [inline examples/metal-nimf-2-port-connection/versions.tf:6]: Please update this to 1.36.4. There's a breaking API response for any versions before that in UAT1 and above.

File: examples/metal-nimf-2-service-profile-connection/versions.tf [hunk 9]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.5.4"
+  required_providers {
+    equinix = {
+      source  = "equinix/equinix"
+      version = ">= 1.34.0"
```
Comment (thogarty) [inline examples/metal-nimf-2-service-profile-connection/versions.tf:6]: Please update this to 1.36.4. There's a breaking API response for any versions before that in UAT1 and above.

**Review by thogarty — CHANGES_REQUESTED** [review]
Requested changes on version numbers to avoid issues in UAT1 and above. Otherwise the change looks good. Thanks, @jkallem-equinix !

File: examples/metal-nimf-2-port-connection/versions.tf (hunk 8 above)
Comment (jkallem-equinix) [reply examples/metal-nimf-2-port-connection/versions.tf]: Done

File: examples/metal-nimf-2-service-profile-connection/versions.tf (hunk 9 above)
Comment (jkallem-equinix) [reply examples/metal-nimf-2-service-profile-connection/versions.tf]: Done

**Review by thogarty — APPROVED** [review]
LGTM. Thank you @jkallem-equinix !

**Pull request merged**